src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -0,0 +1,1153 @@
+        DetailAST parentAst = literalNewAst.getParent();
+        while (TokenUtil.isOfType(parentAst, TokenTypes.LITERAL_NEW, TokenTypes.DOT)) {
+            parentAst = parentAst.getParent();
+        }
+        final DetailAST firstChild = parentAst.getFirstChild();
+        return extractQualifiedName(firstChild);
+    }
+
+    /**
+     * Add non-private instance variables of the super class of the anonymous class
+     * to the variables stack.
+     *
+     * @param obtainedClass super class of the anon inner class
+     * @param variablesStack stack of all the relevant variables in the scope
+     * @param literalNewAst ast node of type {@link TokenTypes#LITERAL_NEW}
+     */
+    private void modifyVariablesStack(ClassDesc obtainedClass,
+            Deque<VariableDesc> variablesStack,
+            DetailAST literalNewAst) {
+        if (obtainedClass != null) {
+            final Deque<VariableDesc> instAndClassVarDeque = classesWithInstVarAndClassVar.get(
+                    obtainedClass.getClassOrInterfaceAst()).getUpdatedCopyOfVarStack(literalNewAst);
+            instAndClassVarDeque.forEach(variablesStack::push);
+        }
+    }
+
+    /**
+     * Checks if there is a class with same name.
+     * Duplicated, will be moved to util after merging
+     * <a>https://github.com/checkstyle/checkstyle/pull/10968</a>
+     *
+     * @param className name of the class
+     * @return true if there is another class with same name.
+     * @noinspection CallToStringConcatCanBeReplacedByOperator
+     */
+    private List<ClassDesc> classesWithSameName(String className) {
+        return classesWithInstVarAndClassVar.values().stream()
+                .filter(classDesc -> {
+                    return classDesc.getQualifiedName().endsWith(
+                            PACKAGE_SEPARATOR.concat(className));
+                })
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * For all classes with the same name as the superclass, finds the nearest one
+     * and registers a subclass for it.
+     * Duplicated, will be moved to util after merging
+     * <a>https://github.com/checkstyle/checkstyle/pull/10968</a>
+     *
+     * @param outerClassName outer class of anonymous inner class.
+     * @param classesWithSameName classes which same name as super class
+     * @return the nearest class
+     */
+    private static ClassDesc getTheNearestClass(String outerClassName,
+            List<ClassDesc> classesWithSameName) {
+        return Collections.min(classesWithSameName, (first, second) -> {
+            int diff = Integer.compare(
+                    classNameMatchingCount(outerClassName, second.getQualifiedName()),
+                    classNameMatchingCount(outerClassName, first.getQualifiedName()));
+            if (diff == 0) {
+                diff = Integer.compare(first.getDepth(), second.getDepth());
+            }
+            return diff;
+        });
+    }
```

Comment:
```
Duplicated, will be moved to util after merging
https://github.com/checkstyle/checkstyle/pull/10968
```
Though I can still move it to util, that will block that PR, there are 5-6 methods like this.